cupy/cuda/memory.pyx

```diff
@@ -556,14 +559,41 @@ cpdef set_allocator(allocator=None):
         allocator = _malloc
     _current_allocator = allocator
 
-
 cpdef get_allocator():
     """Returns the current allocator for GPU memory.
 
     Returns:
         function: CuPy memory allocator.
     """
-    return _current_allocator
+    allocators = getattr(_thread_local, 'allocators', [_current_allocator])
+    return allocators[-1]
+
+@contextlib.contextmanager
```

Comment:
I think [flake8 want another blank line]( https://travis-ci.org/cupy/cupy/jobs/614924320#L292 ) between these functions.

Reply:
good catch,
My vim flake8 plugin is not catching pyx files, I always forget to change it.